### DISCUSSION TITLE
Logo für neue Integration

### DISCUSSION THREAD

**Discussion by emufan** [SukramJ/aiohomematic#117] [Ideas]
Entsprechend https://github.com/danielperna84/hahomematic/discussions/116#discussioncomment-1893907

hier mal als Vorschlag, was für ein Logo man für die Integration nehmen könnte.

HMCloud hat im Brands
![image](https://user-images.githubusercontent.com/22775515/147857090-af9d2630-e1f7-4e7f-aa5e-252d078a3f9b.png)

HM dies hier. Aber wurde mE ja gar nicht verwendet und ist mE auch veraltet.
![image](https://user-images.githubusercontent.com/22775515/147857138-0e672ab2-cce3-4212-b050-3777ffad25e6.png)

Daher mein Vorschlag das hier zu nehmen. Stammt von https://homematic-ip.com/de, habe jedoch das IP rausgenommen, da ja hier nicht nur IP. Habe es genereirt in den Formaten aus dem Brands-Repository, d.h. 256 und 512.

![icon](https://user-images.githubusercontent.com/22775515/147857176-992a307d-eccb-439c-ac73-75079505168c.png)
![icon@2x](https://user-images.githubusercontent.com/22775515/147857178-186a3797-a288-4094-817e-17c0ac6d2f11.png)

**Comment by noxhirsch** [comment]
Hey, ich habe gerade eben diesen Post gesehen ... zufälligerweise habe ich vor ein paar Stunden einen PR im Brands-Repository erstellt, der die HmIP-Icons & -Logos auch für HAHM übernimmt. Diesen PR habe ich jetzt erst mal geschlossen, damit das erst diskutiert werden kann - sorry für meinen Alleingang.

Meine Meinung:
Ich bin für das HmIP-Logo, da es für manche Nutzer Wiedererkennungswert hat, aber inhaltlich "neutral" ist (also ohne IP).

Das Haus taucht bei Homematic nur an einer Position der Website auf, hat also - vor allem ohne IP - IMHO keinen wirklich Wiedererkennungswert. Aus markenrechtlicher Sicht würde ich nicht empfehlen, an einem Logo Änderungen vorzunehmen, es sei denn es ist bekannt, dass der Hersteller keine Probleme damit hat. Also wenn ihr das Haus besser findet, würde ich das IP drin lassen.

Dann ist noch die Frage, was als logo.png verwendet werden soll? Das ist üblicherweise im Querformat, sodass dieses Logo hier eigentlich ideal ist: https://homematic-ip.com/sites/default/files/images/logo-homematic-ip.png

**Reply by noxhirsch** [reply]
Ergänzung: Auf Facebook hat Homematic das Haus auch zeitweise verwendet (etwas aufgehübscht), aber im November durch diese Grafik ersetzt:
<img src="https://user-images.githubusercontent.com/30938717/147879562-c10ba081-145a-4267-9400-06fd0408935b.png" width="200" height="200">

**Reply by emufan** [reply]
Fände ich auch o.k. und hatte ich auch gesehen. Fand bzw. finde nur eine Unterscheidung besser. Daher war ich auf die Suche nach einem anderen Logo auf deren Seite gegangen. 

Ob es o.k. ist, das IP zu entfernen, weiß ich nicht. Fand es nur besser, weil es halt auch ohne IP geht. Und Wiedererkennung ist mE dennoch gegeben, da das ja quasi der Knopf vom IP ist, auf dem auf den Geräten ja auch kein IP ist.

Die x2-Variante von HMCloud und wohl auch die x1 fand ich allerdings auch sehr verwaschen an den Rändern. Daher ebenso ein neu generiertes von mir. Kann das aber mit IP auch wiederholen.

In Summe bin ich aber leidenschaftslos, nur halt besser ein als kein Logo. Und mE auch eher besser ein unterschiedliches zu HMCloud.

**Reply by SukramJ** [reply]
Wenn ihr schon dabei seid:
Gibt es irgendwo Informationen zum Copyright, bzw. ob und welches Logo in welcher Form ... verwendet werden darf?

**Reply by noxhirsch** [reply]
@SukramJ Ich habe jetzt die letzte halbe Stunde dazu recherchiert, aber konnte leider keine Informationen finden.

**Reply by emufan** [reply]
Wer hat denn die alten von HM und HMCloud ins Repo bekommen und kann ggf. berichten, wie dort mit solch Fragen umgegangen ist? ImgBotApp war es sicher nicht allein. 😂

Ich war ehrlich gesagt erstaunt, wie viele Brands da drin sind und bezweifle, dass das alles abgeklärt ist. Aber klar, besser, wenn es konform ist.

**Reply by emufan** [reply]
Hab auch nochmal weitergesucht. Das HMCloud-Icon würde ich defintiv nicht nehmen. Das ist ja sogar das App-Icon von HMCloud und mit der Cloud hat das hier ja nun nüscht zu tun und beide Integrationen mit gleichem Logo zu haben verwirrt mMn nur.

Also doch das obige Häuschen mit oder ohne IP oder den IP-Text, die ich persönlich weniger gut finde. Aber über Geschmack lässt sich ja streiten.

**Reply by noxhirsch** [reply]
Wenn wir das Haus für das Icon (quadratisch) nehmen, was wäre dein (@emufan) Vorschlag für das Logo (Querformat)? Laut Brands Repository sollte ja idealerweise beides vorhanden sein (das Logo ist dann auf der jeweiligen Device-Seite und auf home-assistant.io/integrations zu sehen).

Idee: Aktuell heißt die Integration "Home Assistant Homematic(IP)". Das "Home Assistant" ist in HA & HACS redundant - wenn wir hier schreiben "Homematic(IP) Local", "Homematic(IP) CCU" oder ähnlich, würde das die eventuellen Verwirrungen reduzieren. 
Im Prinzip ist es ja nicht relevant ob IP oder nicht IP, sondern nur ob CCU (in welcher Art auch immer) oder Cloud. Das könnte man entweder über das Logo oder eben einfacher über den Titel rüberbringen.

**Reply by SukramJ** [reply]
Der Name der Integration sollte definitiv noch mal überdacht werden.
Das Backend heist ja `hahomematic` und die custom_component meldet sich als `Home Assistant Homematic(IP)`. Der Ordner in custom_components heist `hahm`. Also macht euch doch auch hierzu mal Gedanken. Das sollte auf jedenfalls vorm Einchecken der Logos geklärt sein.
Bei eQ-3 hab ich auch an den sonst üblichen Stellen nichts zur Logo-Verwendung gefunden.

**Reply by emufan** [reply]
> Wenn wir das Haus für das Icon (quadratisch) nehmen, was wäre dein (@emufan) Vorschlag für das Logo (Querformat)?

Vielleicht da sogar das alte von [Homematic](https://www.home-assistant.io/integrations/homematic/) behalten?

> Idee: Aktuell heißt die Integration auf HA "Home Assistant Homematic(IP)". Das "Home Assistant" ist in HA & HACS redundant - wenn wir hier schreiben "Homematic(IP) Local", "Homematic(IP) CCU" oder ähnlich, würde das die eventuellen Verwirrungen reduzieren.
> Im Prinzip ist es ja nicht relevant ob IP oder nicht IP, sondern nur ob CCU (in welcher Art auch immer) oder Cloud. Das könnte man entweder über das Logo oder eben einfacher über den Titel rüberbringen.

Dachte diese Custom-Version im HACS ist eh nur temporär und heißt so, damit man weiß, wofür das eine Custom Integration ist und würde später sowieso wieder **Homematic** heißen?

**Reply by noxhirsch** [reply]
In der CCU ist das Logo links oben das hier:
![image](https://user-images.githubusercontent.com/30938717/147886037-a5dfd07d-0abf-4e6d-ae3c-98a9491d2fd7.png)
Das wäre vielleicht nicht schlecht für das Querformat (nur ohne die Hintergrundfarbe). Das würde auf jeden Fall hervorheben, dass beide "Systeme" kompatibel sind.
EDIT: Meine Layoutfassung:
![image](https://user-images.githubusercontent.com/30938717/147886300-a666b313-82b9-4fae-8ad8-0da390244de4.png)

**Reply by noxhirsch** [reply]
Falls ihr die Idee mit "Homematic(IP) Local/CCU" sinnvoll finden solltet, würde ich die Integration entsprechend homematicip_local oder homematicip_ccu nennen (oder was auch immer hinter Homematic steht)

**Reply by SukramJ** [reply]
Den Ordner würde ich erstmal bei hahm belassen, solange es eine custom_component ist. So heist auch der verwendete Domain-Name. Wenn man den irgendwann mal den Weg ins HA Repo geht, wird sich der Name sowieso noch mal ändern.
Wichtig ist also aktuell der Name im Config Flow , der auch auf der Integrationsseite angezeigt wird.
Im Brands repo geht es, glaube ich, erstmal um der Ordnernamen, also hahm. Oder?

Ich mag das Logo mit hellem Hintergrund. Blau sieht nach CCU-WebUI aus.

**Reply by SukramJ** [reply]
`Homematic(IP) CCU` gefällt mir.

**Reply by noxhirsch** [reply]
Im Brands Repo ist die Domain relevant - also hahm, richtig.
Das mit dem blauen Hintergrund war auch nur ein Snapshot aus der CCU als Referenz.

Hier auch mal noch ein paar Icon-Ideen, die optisch zu dem Querformat-Logo passen (jeweils unten die Variante für den Darkmode):
![image](https://user-images.githubusercontent.com/30938717/147886669-c1237f04-5fb4-4c61-bdb6-f370bbc1d246.png)

**Reply by noxhirsch** [reply]
Bzw. so geht es auch im Darkmode - ist vielleicht etwas hübscher. Und jetzt auch mit der Schriftart von Homematic 
![image](https://user-images.githubusercontent.com/30938717/147886928-062e125e-9464-4612-a915-18b6da2a07ad.png)
Bei den Icons finde ich CCU auf jeden Fall besser als local.

**Reply by emufan** [reply]
> EDIT: Meine Layoutfassung:

Ich war vorhin auch schon parallel dabei. Hier mein Vorschlag in einer statt drei Farben. Das in der CCU ist ja auch in einer. Wäre auch schon ohne Ränder ind dern Brands-Größen und Namen
![logo](https://user-images.githubusercontent.com/22775515/147886939-007c7926-0da0-4f0f-9096-3f6f04e495ed.png)
![logo@2x](https://user-images.githubusercontent.com/22775515/147886941-318250d1-bedb-4cba-8bb8-5dd12b55955d.png)
.

**Reply by noxhirsch** [reply]
Ich habe auch überlegt, alles in grau zu machen - allerdings hat das dann den Nachteil, dass es etwas trist aussieht und später in der Integrations-Liste optisch etwas untergeht, oder nicht? (unten sind die Rundungen von O, E & C übrigens etwas abgeschnitten)

**Reply by danielperna84** [reply]
Also ganz in grau finde ich nicht so gut. Die Farben sind ja auch Bestandteil der Marke und haben ihren Wiedererkennungswert. Da finde ich das zwei-zeilige von oben für das breite Format am besten. Wie das gut zum Darkmode passt weiß ich nicht. Spontan sehe ich das aber vor allem für die Doku-Seite als relevant an, und die ist (bisher) ja noch mit weißem Hintergrund. Und hier könnte man ja zur Not auch den weißen Hintergrund belassen und 2-3 Pixel im HM-Blau als Rahmen drum packen. Dann hat man immer noch ein weißes Logo, was dank dem Rahmen aber nicht mehr so blöd auf dunklem Hintergrund aussieht.

Beim quadratischen Logo in der Integrationsliste ist mein Bauchgefühl, dass da eigentlich das blaue Quadrat ohne Inhalt (also auch ohne _IP_) den Zweck erfüllt, wenn der Text direkt daneben eindeutig sagt "Homematic (IP) Local" oder so. Damit würde man sich dann nicht zu weit vom _echten_ Logo entfernen.

**Reply by noxhirsch** [reply]
@danielperna84  Mit einem dunklen Hintergrund sieht das bunte Logo auch nicht schlecht aus - aber da man für den Darkmode auch eine extra Logovariante hochladen kann, könnte man hier auch eine weiße Variante nehmen. So macht es Homematic selbst auch: Buntes Logo auf weißem Hintergrund - und wenn der Hintergrund dunkler ist, ist das Logo weiß.
![image](https://user-images.githubusercontent.com/30938717/147888022-483a294d-8b77-4809-aea4-a2a0de9d02d8.png)
![image](https://user-images.githubusercontent.com/30938717/147888057-57b1f65a-bc8f-441f-a74a-646de5c3e328.png)
Als Icon meinst du so?
![image](https://user-images.githubusercontent.com/30938717/147888090-283acde0-cda7-4c7c-83b6-c155f742d025.png)
Schlicht und einfach - gefällt mir. 
Für den Darkmode würde ich es gleich wie oben machen - entweder beides bunt lassen oder weiß (eher letzteres damit es näher am CI von Homematic ist).
EDIT: Das Icon würde ich auf jeden Fall im Darkmode bunt lassen - unabhängig vom Logo. Sieht in der Übersicht besser aus:
![image](https://user-images.githubusercontent.com/30938717/147888658-d365bdd3-b649-4488-b7e4-baeb1f24c46c.png)

Was sollen wir als Namen nehmen? @SukramJ fand `CCU` besser, @danielperna84 wenn ich das richtig interpretiere `Local`. Wenn wir es nicht ins Icon reinschreiben finde ich `Local` etwas hübscher - CCU ist etwas "technischer", dafür natürlich eindeutiger.

**Reply by SukramJ** [reply]
Local ist Ok

**Reply by danielperna84** [reply]
1. Bunt für normal und weiß für Darkmode finde ich super so. 👍
2. Die obere Zeile für Darkmode und normal finde ich gut. Hier würde ich im Darkmode dann doch die Farbe drin lassen, weil einfach nur ein weißes Quadrat ist schon ziemlich abstrakt. Da ist die spezifische Farbe dann wenigstens ein Indikator um welches Produkt es geht.

Ich finde _Local_ grenzt sich rein von der Bedeutung des Wortes besser zur alternativen _Cloud_ Integration ab. Letztere wird es ja voraussichtlich auch weiterhin noch geben. Da würde _CCU_ eher Sinn ergeben, wenn die andere _HomematicIP Access Point_ heißen würde (sollte sie das nicht vielleicht sogar?).

Unabhängig davon finde ich _Local_ als generischen Überbegriff eigentlich gut, da er technisch so gesehen alles abdeckt. Bei _CCU_ könnten dann wieder Fragen kommen wie "Geht das auch mit Raspberrymatic" oder "Was ist mit Homegear" oder "Wo ist hier das Homematic Add-on einzuordnen". In dem Fall würde _Local_ ganz klar sagen: "Wenn es in deinem Netzwerk ist, dann ist _Local_ das richtige für dich". (Mal außenvor gelassen, dass sich der AP physisch im Netzwerk befindet)

**Reply by noxhirsch** [reply]
@danielperna84 Zu Punkt 2 habe ich oben schon einen Edit eingefügt - da sehe ich das genauso wie du - die Farbe muss im Icon bleiben, sonst erkennt man es nicht.
Mit Local hast du Recht - das spart sicherlich einige Fragen 👍 

Dann wäre das die finale Zusammenstellung der Logos & Icons:
![image](https://user-images.githubusercontent.com/30938717/147888951-73a795ff-6512-42d4-a434-6923e9721762.png)
![image](https://user-images.githubusercontent.com/30938717/147888954-380403f2-7408-4cb8-95ef-77917c06956f.png)
![image](https://user-images.githubusercontent.com/30938717/147888958-3934bb72-ce86-4f82-9999-a4531a51b17c.png)
Letzteres nochmal als Beispiel in "echter" Umgebung:
![image](https://user-images.githubusercontent.com/30938717/147889042-f434a6e6-b4de-4f5d-bcc8-e2425ff67918.png)

Wenn das für euch passt, würde ich morgen die Assets nochmal sauber erstellen und ins Brands repo einreichen. Soll ich die Grafiken auch in hahomematic oder custom_homematic nochmal ablegen? Wenn ja, wo? (z.B. custom_homematic Unterordner logos oder images)

**Reply by noxhirsch** [reply]
Für den Namen könnte ich auch einen PR erstellen, wenn es bei "Homematic(IP) Local" bleiben soll, aber ich denke mal, dass @SukramJ den viel schneller beim nächsten version bump in einem Rutsch mit ändern kann.

**Reply by emufan** [reply]
Ich finde es auch gut so. Nur die Ränder/Abstände zu allen vier Seiten sollten gleich sein. Oder ganz weg, zumindest haben die meisten Icons und Logos, die ich mir angeguckt hatten gar keine.

**Reply by danielperna84** [reply]
Ich denke mal in der _finalen_ Version werden die Ränder schon passen, siehe Aussage "Assets nochmal sauber erstellen".

Die Logos würde ich beim custom_homematic dazu packen. Gibt's da was Vektorbasiertes (SVG oder so) damit man hinterher ggf. noch mal daran basteln kann?

**Reply by noxhirsch** [reply]
@emufan: Der Einfachheit halber hatte ich einfach mit dem Snipping Tool Screenshots gemacht und hier eingefügt - ging schneller als jedes Mal eine Version aus Photoshop bzw. Illustrator zu speichern - deshalb sind die Ränder immer unterschiedlich 😬 In der finalen Version werden die Logos & Icons keine Ränder haben (so ist die Anforderung im Brands Repository).

@danielperna84: Das wäre mein Plan gewesen, hier SVGs abzulegen, da ins Brands Repo nur PNGs dürfen und das nicht so wirklich zukunftsfähig ist.

**Reply by noxhirsch** [reply]
@emufan @danielperna84 @SukramJ 
Die Logos sind fertig - wenn das für euch so passt, reiche ich die ins Brands Repo ein.
Hier der PR mit den Grafiken als PNG & SVG: https://github.com/danielperna84/custom_homematic/pull/71

**Reply by emufan** [reply]
Sehe prima aus. 👍 Mag zwar persönlich nicht drei Farben und insb. 2 verschiedene Blautöne, aber das hat ja EQ3 verbockt. 😂

**Reply by noxhirsch** [reply]
Die Logos sind jetzt ins Brands Repository gemerged - kann aber etwas dauern, bis sie in Home Assistant auftauchen (Zitat von Franck auf eine ältere Nachfrage einer anderen Person: "It can take up to 3 weeks before it is visible, due to heavy caching.")
